magento/magento2
# addProductsToCart mutation reports proper errors in user_errors request in GraphQL

### Description (*)

As per this issue, the condition for failed sku, we need to show the error in user_errors field, but in the code, it was pushed the cart error along with the failed user input, so the correct sku also got an error.

### Related Pull Requests
1. Fixes for #37908

### Fixed Issues (if relevant)

1. Fixes magento/magento2#37908

### Manual testing scenarios (*)

1. Create a product "PRODUCT-1" with inventory of 190
2. Create a product "PRODUCT-2" with inventory of 123 (make sure indexers etc. have ran)
3. Create an empty cart:
4. Add a product with for quantity of 123 to the cart.
5. Run this mutation again, you will see an error on user_errors, this is correct.
6. Change the quantity of PRODUCT-2 to 50 in the backend
7. Add the first product to the cart with quantity of 1

We have added a PRODUCT-2 with more than qty 123.
Here We are getting errors it is expected.
user error will come only when the user is trying to add insufficient qty product.

<img width="1308" alt="Screenshot 2023-09-27 at 4 32 10 PM" src="https://github.com/magento/magento2/assets/104893739/c26d9fef-21c7-4f50-9f9b-2a6fcbf45e3f">
<img width="1242" alt="Screenshot 2023-09-27 at 4 31 55 PM" src="https://github.com/magento/magento2/assets/104893739/abbbdf66-aed4-4218-9f67-020785f15d5c">


### Questions or comments


### Contribution checklist (*)
